uni/lib/view/restaurant/widgets/restaurant_page_card.dart

```diff
@@ -43,13 +47,50 @@ class CardFavoriteButton extends StatelessWidget {
       builder: (context, restaurantProvider) {
         final isFavorite =
             restaurantProvider.favoriteRestaurants.contains(restaurant.name);
+        final favoriteCardTypes =
+            Provider.of<HomePageProvider>(context).favoriteCards;
         return IconButton(
           icon: isFavorite ? Icon(MdiIcons.heart) : Icon(MdiIcons.heartOutline),
-          onPressed: () => restaurantProvider.toggleFavoriteRestaurant(
-            restaurant.name,
-          ),
+          onPressed: () => {
+            restaurantProvider.toggleFavoriteRestaurant(
+              restaurant.name,
+            ),
+            if (!isFavorite &&
+                !favoriteCardTypes.contains(FavoriteWidgetType.restaurant) &&
+                restaurantProvider.favoriteRestaurants.length < 2)
+              showRestaurantCardHomeDialog(context, favoriteCardTypes),
+          },
         );
       },
     );
   }
+
+  void showRestaurantCardHomeDialog(
+    BuildContext context,
+    List<FavoriteWidgetType> favoriteCardTypes,
+  ) {
+    showDialog<void>(
+      context: context,
+      builder: (context) => AlertDialog(
+        title: Text(S.of(context).restaurant_main_page),
+        actions: <Widget>[
+          ElevatedButton(
+            onPressed: () {
+              Navigator.of(context).pop();
+            },
+            child: Text(S.of(context).no),
+          ),
+          ElevatedButton(
+            onPressed: () {
+              favoriteCardTypes.add(FavoriteWidgetType.restaurant);
```

Comment:
This more of an optimization thing, but this line might not be needed, since we are updating the provider value below and it should trigger an entire widget rebuild anyway